packages/core/src/types.ts

```diff
@@ -59,12 +59,56 @@ export interface Position {
 // Config
 //------------------------------------------------------------------------------
 
-export type RuleSeverity = 0 | 1 | 2 | "off" | "warn" | "error";
+/**
+ * The human readable severity level used in a configuration.
+ */
+export type SeverityName = "off" | "warn" | "error";
+
+/**
+ * The numeric severity level for a rule.
+ *
+ * - `0` means off.
+ * - `1` means warn.
+ * - `2` means error.
+ *
+ */
+export type SeverityLevel = 0 | 1 | 2;
+
+/**
+ * The severity of a rule in a configuration.
+ */
+export type Severity = SeverityName | SeverityLevel;
+
+/**
+ * Represents the configuration options for the core linter.
+ */
+export interface LinterOptionsConfig {
+	/**
+	 * Indicates whether or not inline configuration is evaluated.
+	 */
+	allowInlineConfig?: boolean;
```

Comment:
```suggestion
	noInlineConfig?: boolean;
```

The config option is `noInlineConfig`.

https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L555-L559